### PR TITLE
fix: handle JSON Objects in parseFromFluxResults

### LINKED ISSUE #3637: Bug Duty: Table and SimpleTable break when column is a nested object
## About the bug

When the column value is a nested object, `parseFromFluxResults` doesn't work correctly. This is seen in `SimpleTable` as well as `Table` graph. 

image

### LINKED ISSUE #3736: Dashboards - markdown editor - Emoji, strikethrough
## About the bug

**Steps to reproduce:**
1. Create a dashboard
2. Add a note
3. Use markdown editor
4. Type any :emoji:
5. Use tilde symbols to add strikethrough

**Expected behavior:**
image
image

**Actual behavior:**
image

## About your environment

**Environment info:**
Testing with remocal on a UI project (Ubuntu 20.04)

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes #3637 

In https://github.com/influxdata/ui/pull/4700, we handled a special case where the data in the column can be a string of `CSV` data. 

In this one, we handle the case when the data is a `JSON` object. 

The approach is to take the JSON string, remove all the double quotes and replace them with single quotes, (this saves the csvParser from being thrown off) and then wrap the JSON object in double quotes. 

Example
-> `{"key" : "value"}` came in from the flux query
-> `{'key': 'value'}` : we replace double quotes with single quotes
-> `"{'key': 'value'}"` : we wrap the string form of the JSON object in double quotes so we can display it.

----

### Proof of it working:

Please look at the `_value` column in all the screenshots for the demo of JSON and CSV strings working.

**Simple CSV in Simple Table:** 

Screen Shot 2022-06-02 at 4 01 23 PM

Screen Shot 2022-06-02 at 4 05 17 PM

**Simple String in Simple Table:** 

Screen Shot 2022-06-02 at 4 04 22 PM

**JSON in Simple table:** 

Screen Shot 2022-06-02 at 4 01 18 PM

Screen Shot 2022-06-02 at 4 06 50 PM

**Complex result with multiple JSON objects and normal strings**

Screen Shot 2022-06-02 at 4 01 09 PM

### CODE REVIEW

File: src/timeMachine/utils/rawFluxDataTable.ts [hunk 1]
```diff
@@ -29,6 +29,18 @@ export const parseFromFluxResults = (
   const groupSet = new Set(fluxGroupKeyUnion)
   let max = 0
 
+  // checks whether the string is valid JSON object or not
+  const isJsonObject = jsonString => {
+    try {
+      const object = JSON.parse(jsonString)
+      if (object && typeof object === 'object') {
+        return object
```
Comment (hoorayimhelping) [inline src/timeMachine/utils/rawFluxDataTable.ts:37]: I think this should `return true`. An object is truthy, but it's good to be explicit in these cases and not rely on weird JS truthiness rules.

File: src/timeMachine/utils/rawFluxDataTable.ts [hunk 2]
```diff
@@ -29,6 +29,18 @@ export const parseFromFluxResults = (
...
+      }
+    } catch (_) {}
```
Comment (hoorayimhelping) [inline src/timeMachine/utils/rawFluxDataTable.ts:39]: I think you can just do `catch {}`

File: src/timeMachine/utils/rawFluxDataTable.ts [hunk 3]
```diff
@@ -72,7 +84,16 @@ export const parseFromFluxResults = (
...
       }
-      if (typeof columnData === 'string') {
+
+      // (Sahas):
+      // the columnData can have a comma (,) in two cases
+      // 1. it's a JSON Object
+      // 2. It's a string of CSV
+      if (isJsonObject(columnData)) {
+        // replace Double quotes \" with single quotes \' in a json object
+        // columnData = `"${columnData.replace(/['"]+/g, '\'')}"`
```
Comment (hoorayimhelping) [inline src/timeMachine/utils/rawFluxDataTable.ts:94]: this line can be ✂️, right?

**Review by hoorayimhelping — COMMENTED** [review]
A couple of suggestions, let me know what you think (and request a re-review when you're ready for one)

File: src/timeMachine/utils/rawFluxDataTable.ts (hunk 2 above)
Comment (ChitlangeSahas) [reply src/timeMachine/utils/rawFluxDataTable.ts:39]: ✅

File: src/timeMachine/utils/rawFluxDataTable.ts (hunk 1 above)
Comment (ChitlangeSahas) [reply src/timeMachine/utils/rawFluxDataTable.ts:37]: Got it, fixed!

File: src/timeMachine/utils/rawFluxDataTable.ts (hunk 3 above)
Comment (ChitlangeSahas) [reply src/timeMachine/utils/rawFluxDataTable.ts]: Yes, fixed 👍

**Review by hoorayimhelping — APPROVED** [review]

**Pull request merged**